src/network/clientpackethandler.cpp

```diff
@@ -562,6 +562,10 @@ void Client::handleCommand_HP(NetworkPacket *pkt)
 
 	u16 hp;
 	*pkt >> hp;
+	bool damage_effect = true;
+	try {
+		*pkt >> damage_effect;
+	} catch (PacketError &e) {};
```

Comment:
did you test that this works? I'm not sure whether reading bools from packets throws exceptions (`readU8` does not for example)

Reply:
```c++
	// Wrap transition_time extraction within a
	// try-catch to preserve backwards compat
	try {
		*pkt >> transition_time;
	} catch (PacketError &e) {};
```
is used for `f32`, why wouldn't it work for bools? Similarly sound code uses

```c++
	bool ephemeral = false;

	*pkt >> server_id >> name >> gain >> type >> pos >> object_id >> loop;

	try {
		*pkt >> fade;
		*pkt >> pitch;
		*pkt >> ephemeral;
	} catch (PacketError &e) {};
```